packages/mattermost-redux/src/actions/channels.ts

```diff
@@ -1428,6 +1433,27 @@ export function actionsToMarkChannelAsUnread(getState: GetStateFunc, teamId: str
     return actions;
 }
 
+export function markMostRecentPostInChannelAsUnread(channelId: string): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let state = getState();
+        let postId = getMostRecentPostIdInChannel(state, channelId);
+        if (!postId) {
+            await dispatch(getPosts(channelId));
+            state = getState();
+            postId = getMostRecentPostIdInChannel(state, channelId);
+            if (postId) {
+                const lastPost = getPost(state, postId);
+                dispatch(markPostAsUnread(lastPost, 'CENTER'));
+            }
```

Comment:
We can actually remove this block since it'll this'll exit the if block and still hit the one below it. As written, this currently marks a channel as unread twice if the channel isn't loaded